### PR TITLE
Test case for OSPF with BFD

### PR DESCRIPTION
**Pull request by Ghulam-Bahoo**
### Description of PR
This PR introduces a new test case for testing [BFD with OSPF](https://github.com/sonic-net/sonic-mgmt/pull/11899).
Summary:
This PR tests the functionality of BFD with OSPF. Firstly, OSPF neighborship is established between DUT and the neighboring devices and then BFD is instantiated on all devices in the topology. Lastly, BFD is enabled on all interfaces between DUT and the neighboring devices.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
To test the functionality of BFD protocol in conjunction with OSPF. BFD provides fast convergence time in response to link failure reducing overall network downtime.
#### How did you do it?
By enabling BFD on all devices in the topology.
#### How did you verify/test it?
By simulating a link failure(one of the interfaces between DUT and the neighboring device was brought down) and verifying how fast the traffic is shifted to another available path.z

### CODE REVIEW

**Comment by Ghulam-Bahoo** [comment]
@yxieca can you please assign a reviewer for this PR?

**Comment by yxieca** [comment]
> @yxieca can you please assign a reviewer for this PR?

@StormLiangMS  can help, but I think it is better to ask for help from community OSPF/BFD experts.

**Comment by ridahanif96** [comment]
@yxieca Hi Ying, do you know currently who in community dealing with OSPF/BFD, so we can ask them to help review as well.

**Comment by asraza07** [comment]
@yxieca Hi Ying, We have added the Sanity Tests result as well. Can you pls review. Thanks in advance!

File: tests/ospf/conftest.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    for nbr_name, nbr_info in nbrhosts.items():
+        nbr_ip = setup_info["nbr_addr"].get(nbr_info["ip"])
+        if nbr_ip:
+            configure_ospf_and_bfd(
+                nbr_info["host"],
+                nbrhosts[nbr_name].hostname)
+            setup_info["nbr_addr"][nbr_ip] = {"hostname": nbr_info["hostname"]}
+
+    setup_info["int_info"] = get_ospf_neighbor_interface()
```
Comment (auspham) [inline tests/ospf/conftest.py:31]: Hi @Ghulam-Bahoo  /@asraza07  just checking if this test is passing from your end? I have trouble running from my side, some of the method/arguments are not called correctly for example this one here is missing argument `host`.

I tried to fix myself but eventually could not get the OSPF established

**Pull request merged**